### ISSUE TITLE
Move Down/Next: Changed behavior

### ISSUE THREAD

**Issue by tvarsis** [anametologin/krohnkite#93]
**Symptom**
Updated to  0.9.8.2 and that changed the be behavior of Move Down/Next and Move Up/Prev. After the upgrade the window just stop at the bottom/top, where as in prior release you could just keep hitting for them to circle around the full layout (clockwise or counter clockwise and left/right/up/down). The old behavior was much better and would reduce the number of shortcuts needed to cycle a window to the left/right side by just keep hitting the same shortcut.

Is there a reason for this change or is this a bug? 
If this is intended, then could we please get a new shortcut with the old logic in it?

**Environment**
 - Distro: OpenSUSE Tumbleweed
 - Krohnkite version: 0.9.8.2

**Comment by anametologin** [comment]
> The old behavior was much better
Are you talk about Columns layout? If you do then the behavior wasn't changed. Rotate on 90 degrees and you get 0.9.8.1, you can also config the initial rotation angle in settings.

**Comment by tvarsis** [comment]
So if I have a layout with 1 window to the left taking up 50% of screen in the first column and then on the second column I have 3 windows in 3 rows taking up the other half of the screen. If i then mark the top window in the right column and click the shortcut to Move Down/Next 3 times, it would with the old behavior move that window to be the big window on the left taking up the full first column/half of the screen. If I press a 4th time, it would then go back to the top of the right half of the screen and keep on rotating clockwise for every press. With new behavior, after to press on the shortcut it stops at the bottom of the right column and nothing else happens. It never continue to move to the left side of the screen.

Same applies to the Move Up/Prev flow, but counter-clockwise where it gets stuck in the top position and never moves to the left column as per old logic did. Not sure exactly which version I had before upgrading, it was probably like 6 month old or so.

**Comment by anametologin** [comment]
> it would with the old behavior move that window to be the big window on the left taking up the full first column/half of the screen. If I press a 4th time, it would then go back to the top of the right half 

In Columns layout If you want move a window to the left you have to push one time left but not 4 times down or up.

May be I don't understand you, video or screenshot would help me.

**Comment by tvarsis** [comment]
@anametologin Yes, but you didn't need to do that before. That is my point. Before they wrapped around which is a much nicer workflow imo since it requires less shortcuts and usually you don't have that many window open, meaning you can use the same shortcut for both. Makes sense?

**Comment by anametologin** [comment]
> imo since it requires less shortcuts and usually you don't have that many window open, meaning you can use the same shortcut for both. Makes sense?

I don't know, but this isn't `the old behavior` in Krohnkite. I haven't touched this behavior in other layouts but in Tile layout I don't see this behavior either.

**Comment by tvarsis** [comment]
@anametologin Hmm, well that is how it has been working for me for like 6 month and it stopped after the upgrade. So something is at least different :smile: 

I also saw this in the release notes of 0.9.8:

-support moving windows left/right up/down by krohnkite shortcuts

Which sounds like changes have been made in that logic, no?

**Comment by tvarsis** [comment]
@anametologin Ok,thanks! Maybe I can put in a feature request then to have a type of shortcut to work as I described? Would be a great addition imo.

**Comment by tvarsis** [comment]
@anametologin Sorry, just another point around this. Reading the naming and expected result based on it:

Focus Next: Pressing this multiple times goes through all windows on the screen
Focus Previous: Same as above, but other direction
Move Right: Only move window to right side, multiple presses does nothing
Move Left: Same as above but left
Move Down/Next: Only move window down, stops at bottom
Move Up/Prev: Same as above, stops at top

The naming of Down/Next and Up/Prev indicates to me that it was indeed intended to work as it did prior, else it would just be named Down and Up and not include those Next/Prev part which to me means it should continue to move to next position even when it hits the top/bottom on one side, same as the Next/Prev for Focus shortcut does. Would you not agree to that?

**Comment by anametologin** [comment]
The behavior of keyboard shortcuts depends on the layout, so there is no point in discussing this in general terms.
